webpack.js

```diff
@@ -0,0 +1,30 @@
+const path = require('path')
+const webpack = require('webpack')
+const webpackConfig = require('@nextcloud/webpack-vue-config')
+
+webpackConfig.entry = {
+	text: path.join(__dirname, 'src', 'main.js'),
+	files: path.join(__dirname, 'src', 'files.js'),
+	public: path.join(__dirname, 'src', 'public.js'),
+	viewer: path.join(__dirname, 'src', 'viewer.js'),
+}
+
+webpackConfig.output.chunkFilename = '[id].js?v=[contenthash]'
+
+webpackConfig.optimization.chunkIds = 'named'
+webpackConfig.optimization.splitChunks.cacheGroups = {
+	defaultVendors: {
+		test(module) {
+			return module.resource && module.resource.includes(`${path.sep}node_modules${path.sep}`) &&
+				!module.resource.includes(`${path.sep}highlight.js${path.sep}`)
+		},
+		name: 'vendors',
+	}
+}
```

Comment:
What does this do?

Reply:
it makes sure that vendor files get into a separate chunk, except the highlight js ones which should stay at one chunk per supported syntax highlighting language.

Reply:
What benefit do we get from this?

Reply:
Vendors are nice to split as long as they don't get all used at the same time, right?
Like dynamic loading?
But if we always end up loading 90% of the chunks, we just create more bandwith and doesn't help cleaning the bundles imho :)

Reply:
The main issue i have is that I fail to tell webpack to only split off the highlight.js chunks and the ones that are dynamically imported (e.g. for only loading the files app init code and not the whole viewer component right away). Therefore I went for that hybrid approach that unfortunately leaves one extra chunk.

Reply:
I'm happy to look into that into a follow up again :wink:

Reply:
You can do that with an ignore plugin and do a dynamic import of the highlight lib :)
https://github.com/nextcloud/contacts/blob/017026a8e42717cc8e1935af5ce61aa46b3e9f79/webpack.js#L8

Reply:
Iirc, any direct runtime import will overrule compile time bundling